tests/util/checker/checker.go

```diff
@@ -53,6 +59,11 @@ func Check(paths []string, factory RulesFactory, whitelist *Whitelist, report *R
 
 // fileCheck checks a file using the given rules, and write to the given Report.
 func fileCheck(path string, rules []Rule, whitelist *Whitelist, report *Report) {
+	// skip over tests
+	if IgnoreTestLinterData && strings.Contains(path, "testlinter/testdata") {
```

Comment:
This doesn't look like the right, final choice. A config plumbed in up top to control where to traverse is probably the better approach. Perhaps a bug and an associated TODO to handle this?

Reply:
Added a TODO. I agree that this needs further work, for a more consistent approach to skipping linter tests in a linter. It's probably the only useful capability of gometalinter that we relied upon, and now that it's gone, we need to find another way.